kie-api/src/main/java/org/kie/api/KieServices.java

```diff
@@ -168,6 +168,22 @@
      */
     KieContainer newKieClasspathContainer(String containerId, ClassLoader classLoader);
 
+    /**
+     * Creates a new KieContainer for the classpath using the given classLoader,
+     * regardless if there's already an existing one,
+     * enforcing the given containerId.
+     * <p>If enforcing the containerId is successful, the KieContainer and its containerId are registered within the KieServices.
+     * Deregistration can be performed by calling {@link KieContainer#dispose()} on the resulting KieContainer.</p>
+     * @param containerId a unique containerId (non-null).
+     * @param classLoaderContainer classLoader
+     * @param releaseId release id of this container
+     * @return new kie classpath container
+     * @throws IllegalStateException if the containerId is already existing for another container, and therefore cannot be enforced.
+     */
```

Comment:
it can also throw _UnsupportedOperationException_ for those containers where release id is not supported (backward compatibility)

Reply:
done